programs/marinade-finance/src/instructions/delayed_unstake/order_unstake.rs

```diff
@@ -37,38 +37,36 @@ pub struct OrderUnstake<'info> {
 }
 
 impl<'info> OrderUnstake<'info> {
-    fn check_burn_msol_from(&self, msol_amount: u64) -> Result<()> {
-        if self
+    // returns user msol available balance (can be owner or delegate)
+    fn check_burn_msol_from(&self, msol_amount: u64) -> Result<u64> {
+        let user_msol_balance = if self
```

Comment:
Balance must be amount only not delegation. No reason to log delegation here

Reply:
done